### ISSUE TITLE
The argument 'pwsh' is not recognized as the name of a script file

### ISSUE THREAD

**Issue by Lockszmith-GH** [xpipe-io/xpipe#77]
In my Local System, I was trying to add a custom shell:

* Click on `Add Command ...`
* Selected `Shell command` as the type.
* In Command pasted the path to my installation of PowerShell Core:  
  `"C:\Program Files\PowerShell\7\pwsh.exe"`
* Named the connection

(BTW, clicking Finish here brought up the 'Add connections' wizard dialog - but that's a different issue)

Clicking on the new PowerShell Core entry under Local Machine, terminal opens with a new tab with the following

```console
The argument 'pwsh' is not recognized as the name of a script file. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

Usage: pwsh[.exe] [-Login] [[-File] <filePath> [args]]
                  [-Command { - | <script-block> [-args <arg-array>]
                                | <string> [<CommandParameters>] } ]
                  [-ConfigurationName <string>] [-ConfigurationFile <filePath>]
                  [-CustomPipeName <string>] [-EncodedCommand <Base64EncodedCommand>]
                  [-ExecutionPolicy <ExecutionPolicy>] [-InputFormat {Text | XML}]
                  [-Interactive] [-MTA] [-NoExit] [-NoLogo] [-NonInteractive] [-NoProfile]
                  [-NoProfileLoadTime] [-OutputFormat {Text | XML}]
                  [-SettingsFile <filePath>] [-SSHServerMode] [-STA]
                  [-Version] [-WindowStyle <style>]
                  [-WorkingDirectory <directoryPath>]

       pwsh[.exe] -h | -Help | -? | /?

PowerShell Online Help https://aka.ms/powershell-docs

All parameters are case-insensitive.
Press any key to continue . . .
```

Reviewing the Debug log, I see the following:
> ```log
> 11:05:26:326 - trace: [proc] Starting command execution ... {
>     baseCommand=start "PowerShell Core" wt.exe -w 1 nt --title "PowerShell Core" "C:\Users\Szkolnik\.xpipe\temp\exec-188397175.bat"
>     shellType=cmd
>     elevated=false
>     workingDirectory=null
> }
> ```

content of `C:\Users\Szkolnik\.xpipe\temp\exec-188397175.bat` is:

```bat
@echo off
CHCP 65001 >NUL
cmd /K "C:\Users\Szkolnik\.xpipe\temp\exec-2022897643.bat"
```

content of `C:\Users\Szkolnik\.xpipe\temp\exec-2022897643.bat` is:

```bat
@echo off
CHCP 65001 >NUL
TITLE PowerShell Core
set TERM=xterm-256color
cd /D "C:\Users\Szkolnik"
"C:\Program Files\PowerShell\7\pwsh.exe" pwsh /NoLogo -ExecutionPolicy Bypass -NoExit -File "C:\Users\Szkolnik\.xpipe\temp\exec-1104102403.ps1"
pause
exit
```

Seems that `pwsh` is somehow pushed as an argument

> "C:\Program Files\PowerShell\7\pwsh.exe" `pwsh` /NoLogo -ExecutionPolicy Bypass -NoExit -File "C:\Users\Szkolnik\.xpipe\temp\exec-

**Comment by crschnick** [comment]
Yeah the custom shell command functionality is not very polished.

So if I understand that correctly, you had to use that because your pwsh is not present in the path? Because if it is you could just use a shell environment for that.

**Comment by Lockszmith-GH** [comment]
It is actually in the path, I just tested, but even if all I supply is `pwsh` as the command, the command line will be:

```
pwsh pwsh /NoLogo -ExecutionPolicy Bypass -NoExit -File "C:\Users\Szkolnik\.xpipe\temp\exec-...
```

**Comment by crschnick** [comment]
Okay but this does not work for you?
![image](https://github.com/xpipe-io/xpipe/assets/72509152/1f286b22-9081-40e0-bc6c-1eccb15112d7)

The reason it is called twice is that this is an ugly solution for arbitrary shell commands. If you enter `ssh ...` here for example, we would have to explicitly append the shell executable to launch. But in the case of pwsh this actually breaks it. A solution would be to manually check what type of command you actually entered there. But since shell environments can achieve the same functionality, I didn't really pay much attention to the custom shell commands.

**Comment by crschnick** [comment]
Yeah I see how this can be confusing. The summary is that it is very complicated to provide a correct handling of an arbitrary shell command that you can provide without altering it in the background to make it work somehow. I'm thinking of dropping the custom shell command functionality due to this.

Yes I should be able to rework the execution policy, I didn't put much thought into powershell specific features.

**Comment by crschnick** [comment]
So I was able to work on both issues in the [PTB release](https://github.com/xpipe-io/xpipe-ptb)

The custom shell command should now better understand on how to join commands for shells and do not execute `pwsh` twice. Furthermore, I reworked the ExecutionPolicy handling to now dynamically check the current execution policy, change it for the current process if needed to run init scripts, and change it back for the user terminal session.

**Comment by Lockszmith-GH** [comment]
I'll take it for a spin

**Comment by Lockszmith-GH** [comment]
Excellent. I like the design of PTB as well - cleaner. It also seems faster. Thanks for this.

**Comment by crschnick** [comment]
Alright that is good to know that it works for you. You can keep using it, there will be frequent updates for the PTB as well. Having people use new major versions makes my life easier with regards to quality assurance and testing.

**Comment by crschnick** [comment]
Closing this as it is released now in 1.6
